typescript/ws-protocol/src/FoxgloveClient.ts

```diff
@@ -164,9 +164,15 @@ export default class FoxgloveClient {
     this.send({ op: "unsubscribe", subscriptionIds: [subscriptionId] });
   }
 
-  advertise(topic: string, encoding: string, schemaName: string): ClientChannelId {
+  advertise(
+    topic: string,
+    encoding: string,
+    schemaName: string,
+    schema?: string,
+    schemaEncoding?: string,
```

Comment:
Feels like this is too many string params in a row, should become an object

Reply:
I was thinking the same, but didn't want to break API. Will deprecate this method and add one with an object

Reply:
We could also bump to v0.6 and break the API :)